src/js/wall/LoginForm.tsx

```diff
@@ -30,23 +29,32 @@ const LoginContainer = styled.div`
 type LoginFormProps = {
     /** Error message for the login process. */
     error: string;
-    /** Callback to handle login submission */
-    onLogin: (username: string, password: string, remember: boolean) => void;
 };
 
 /** Handles the user login process. */
-export function LoginForm({ error, onLogin }: LoginFormProps) {
+export function LoginForm({ error }: LoginFormProps) {
     const { control, handleSubmit, register } = useForm();
+    const dispatch = useDispatch();
+    const loginMutation = useLoginMutation();
+
+    function onSubmit({ username, password, remember }) {
+        loginMutation.mutate(
+            { username, password, remember },
+            {
+                onSuccess: data => {
+                    dispatch({ type: LOGIN.SUCCEEDED, payload: data });
+                },
```

Comment:
Not suitable action to do the same thing? I am fine with this if not. 

Our general redux pattern is to define actions, but given that this will be removed shortly I will merge as is